test/id.test.js

```diff
@@ -247,3 +247,182 @@ describe('mongodb default id name', function() {
     );
   });
 });
+
+describe('strictObjectIDCoercion', function() {
+  var ObjectID = ds.connector.getDefaultIdType();
+  var objectIdLikeString = '7cd2ad46ffc580ba45d3cb1f';
+
+  context('set to false (default)', function() {
+    var User = ds.createModel(
+      'user1',
+      {
+        id: {type: String, id: true},
+        name: String,
+      }
+    );
+
+    beforeEach(function(done) {
+      User.deleteAll(done);
+    });
+
+    it('should find model with ObjectID id', function(done) {
```

Comment:
Let's use `async` functions instead of callbacks in new tests please.

```js
it('should find model with ObjectID id', async () => {
  const user = await User.create({name: 'John'});
  user.id.should.be.an.instanceOf(ds.ObjectID);
  const found = await User.findById(user.id);
  user.should.not.be.null();
});
```